src/passes/OptimizeInstructions.cpp

```diff
@@ -2225,6 +2255,23 @@ struct OptimizeInstructions
         matches(curr, binary(And, pure(&left), ival(0)))) {
       return right;
     }
+    // -x * C   ==>    x * -C,   if  shrinkLevel == 0  or  C != C_pot
+    // -x * C   ==>   -(x * C),  otherwise
```

Comment:
Right! The idea is not to convert power of two constants (C_pot), because it will then be converted to (x << ctz(C)) and this is only possible if the constant is positive. However, if we are optimizing for size, it doesn't really matter, so we check `shrinkLevel == 0`

Reply:
basically if should be
`x * -C` if (shrinkLevel > 0 || (shrinkLevel == 0 && C != C_pot))
`-(x * C)` otherwise